data/projects/optimize_caching.md

```diff
@@ -1,28 +1,50 @@
-This project involves improving coala's performance by implementing caching
-strategies. One example is to use git-like approach at file-collection level
-to not explore directories that haven't been modified. Caching per section can
-also be implemented so sections without issues don't rerun on all files where
-issues in other sections occurred.
+Currently, coala supports caching of project files. This was a big performance
+improvement over running code analysis for all files, even if most files
+are unchanged since the last run.
 
-Own performance optimization ideas are highly encouraged!
+In this project, you're to improve on the current strategies. Some enhancements
+such as section-level caching, globbing improvements to not expand on
+unmodified directories, and others have been preliminarily explored. Please
+refer to the related issues to learn more about these.
 
+The student is also highly encouraged to come up with newer solutions that may
+increase performance even further. The applicant is allowed to flexibly change
+the milestones depending on how major the newly proposed solutions are, after
+explicit approval from the mentors and admin.
```

Comment:
I think the second sentence is general to all projects of coala, and can be removed.